crates/gen/src/class.rs

```diff
@@ -161,31 +160,37 @@ impl Class {
 
             quote! {
                 #[repr(transparent)]
-                #[derive(::std::clone::Clone, ::std::default::Default, ::std::cmp::PartialEq)]
-                pub struct #name { ptr: ::winrt::ComPtr<#default_name> }
+                pub struct #name(::winrt::Object);
                 impl #name {
                     #new
                     #methods
                     #async_get
                     #call_factory
                 }
+                impl ::std::clone::Clone for #name {
+                    fn clone(&self) -> Self {
+                        Self(self.0.clone())
+                    }
+                }
+                impl ::std::cmp::PartialEq for #name {
+                    fn eq(&self, other: &Self) -> bool {
+                        self.0 == other.0
+                    }
+                }
+                impl ::std::fmt::Debug for #name {
+                    fn fmt(&self, f: &mut ::std::fmt::Formatter<'_>) -> ::std::fmt::Result {
+                        write!(f, "{:?}", self.0)
+                    }
+                }
                 #type_name
-                unsafe impl ::winrt::ComInterface for #name {
-                    type VTable = #abi_name;
-                    const IID: ::winrt::Guid = <#default_name as ::winrt::ComInterface>::IID;
+                unsafe impl ::winrt::Interface for #name {
+                    type Vtable = #abi_name;
+                    const IID: ::winrt::Guid = <#default_name as ::winrt::Interface>::IID;
                 }
                 unsafe impl ::winrt::RuntimeType for #name {
+                    type DefaultType = Option<Self>;
                     const SIGNATURE: ::winrt::ConstBuffer = ::winrt::ConstBuffer::from_slice(#signature);
                 }
-                unsafe impl ::winrt::AbiTransferable for #name {
-                    type Abi = ::winrt::RawComPtr<#default_name>;
-                    fn get_abi(&self) -> Self::Abi {
-                        <::winrt::ComPtr<#default_name> as ::winrt::AbiTransferable>::get_abi(&self.ptr)
-                    }
-                    fn set_abi(&mut self) -> *mut Self::Abi {
-                        <::winrt::ComPtr<#default_name> as ::winrt::AbiTransferable>::set_abi(&mut self.ptr)
-                    }
-                }
                 impl ::std::convert::From<#name> for ::winrt::Object {
                     fn from(value: #name) -> Self {
                         unsafe { ::std::mem::transmute(value) }
```

Comment:
In fact, the `ManuallyDrop` isn't even needed since the `value.0` will simply move the value out and return ownership.

```rust
impl<#constraints> ::std::convert::From<#name> for ::winrt::Object {
    fn from(value: #name) -> Self {
        value.0
    }
}
```

Reply:
Ah yes of course. Cool!